L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleHGCMulticlusters.cc

```diff
@@ -151,6 +161,10 @@ void HGCalTriggerNtupleHGCMulticlusters::fill(const edm::Event& e, const edm::Ev
     cl3d_ntc90_.emplace_back(cl3d_itr->triggerCells90percent());
     cl3d_bdteg_.emplace_back(id_->value(*cl3d_itr));
     cl3d_quality_.emplace_back(cl3d_itr->hwQual());
+    for (auto interp : interpretations_) {
```

Comment:
Maybe the list of available interpretations could be directly retrieved from the cluster object?

Reply:
I have now implemented this...however I am not particularly happy with the fact that this requires a loop for every cluster to map the keys into a vector...maybe we should get some level of caching implemented?

Reply:
An other possibility could be to implement an iterator over the interpretations. This way there is only one loop involved.

Reply:
Ok, I followed the approach of the iterators...although it turned to be more difficult than expected.